### PR TITLE
Oy2-15959: Clear Measure Button

### PR DESCRIPTION
**Pull request by jphechter**
## Purpose

Creates a button only visible in development environments that clears any current measure data. This is primarily intended to be used as a testing tool.

#### Linked Issues to Close

[OY2-15959](https://qmacbis.atlassian.net/browse/OY2-15959)

## Approach

- Create a component that appears in all measure footers containing all action buttons
  - component implemented in MeasureWrapper
  - Measure.Props and specific measure props (FUAAD, IETAD, CPAAD) updated to reflect change
  - Do not show component if measure autocompletes
- A button labeled "Clear Data" is conditionally rendered based on the current environment (if not prod)
- On button click, the user is prompted to confirm they do intend to clear the data
- If yes, an empty object is sent sent to the server effectively resetting the measure and zeroing all fields
- Measure status is set to "incomplete"

We created a net new function `handleClear()`, similar to `handleSubmit()` to accomplish this work. Both call a helper function `submitDataToServer()` that is responsible for assembling the payload and building the API request. Notably, this is where the MeasureStatus is set to "complete". To make this function more flexible and preserve any existing usage, we added an optional parameter `status` with the default value `MeasureStatus.COMPLETE`. The new implementation `handleClear()` utilizes this to reset the status. Arguably, a default shouldn't be the final state, but in the interest of touching the least amount of code with this change, we opted to preserve as much existing functionality as possible.

## Learning

Initially we intended to create a new environment variable `BRANCH_NAME` to do the conditional render check. Somewhere in the build process, that name must be used somewhere else because the value was consistently changed to "local". We found that during the deploy this is actually getting set to the branch name. As long as the current branch _is not_ `prod` the button will render.

## Acceptance Criteria
```
Given that when I am in a measure
When a user clicks onto the clear button 
Then the clear button will clear data from the measure.
Given when I’m in PROD
When a user clicks on a measure
Then clear button shall not be displayed.
```

## Assorted Notes/Considerations
nothing additional

#### Pull Request Creator Checklist

- [X] This PR has an associated issue or issues.
- [X] The associated issue(s) are linked above.
- [X] This PR meets all acceptance criteria for those issues.
- [X] This PR and linked issue(s) are adequately documented
- [X] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [x] Someone has been assigned this PR.
- [x] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

File: services/ui-src/src/components/MeasureWrapper/index.tsx [hunk 1]
```diff
@@ -87,6 +95,20 @@ export const MeasureWrapper = ({ measure, name, year, measureId }: Props) => {
...
+    setShowClearDataModal(true);
+  };
+
+  const handleClearDataModalResponse = (continueClear: boolean) => {
+    setShowClearDataModal(false);
+
+    if (continueClear) {
+      submitDataToServer({ data: {} }, MeasureStatus.INCOMPLETE);
+      console.log("measure cleared");
+      console.log({ errors });
```
Comment (benjaminpaige) [inline services/ui-src/src/components/MeasureWrapper/index.tsx:108]: prob dont need these

File: services/ui-src/src/components/CompleteMeasureFooter/index.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import * as CUI from "@chakra-ui/react";
+import * as QMR from "components";
+
+import config from "config";
+interface Props {
+  handleClear: any;
+  handleSubmit: any;
+  handleValidation: any;
```
Comment (benjaminpaige) [inline services/ui-src/src/components/CompleteMeasureFooter/index.tsx:8]: these should be easy to add types for  - prob just () => void

File: services/ui-src/src/components/MeasureWrapper/index.tsx [hunk 3]
```diff
@@ -187,6 +212,12 @@ export const MeasureWrapper = ({ measure, name, year, measureId }: Props) => {
         handleModalResponse={handleValidationModalResponse}
         bodyText="There are still errors on this measure, would you still like to complete?"
       ></QMR.YesNoModalDialog>
+      <QMR.YesNoModalDialog
+        isOpen={showClearDataModal}
+        headerText="Clear Data"
+        handleModalResponse={handleClearDataModalResponse}
+        bodyText="Do you want to clear all measure data?"
+      ></QMR.YesNoModalDialog>
```
Comment (benjaminpaige) [inline services/ui-src/src/components/MeasureWrapper/index.tsx:220]: i love that you reached for this, and im not sure whats in the ticket but i dont think we need a modal here. its only used by devs and for e2e testing.

File: services/ui-src/src/measures/2021/CPAAD/validation/types.ts [hunk 4]
```diff
@@ -2,8 +2,6 @@ export namespace Measure {
   export interface Props {
     name: string;
     year: string;
-    handleSubmit?: any;
-    handleValidation?: any;
```
Comment (benjaminpaige) [inline services/ui-src/src/measures/2021/CPAAD/validation/types.ts:6]: now i see why you used the any type initially lol

**Review by benjaminpaige — CHANGES_REQUESTED** [review]
love that you broke these buttons out into their own comp. love that we can use them at the measureWrapper level. great job. just a few small things in the comments. Plus... unit tests: wooot!!

File: services/ui-src/src/components/MeasureWrapper/index.tsx (hunk 1 above)
Comment (jphechter) [reply services/ui-src/src/components/MeasureWrapper/index.tsx:108]: Included.

File: services/ui-src/src/components/CompleteMeasureFooter/index.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+          {config.BRANCH_NAME !== undefined && config.BRANCH_NAME !== "prod" && (
+            <QMR.ContainedButton
+              buttonProps={{
+                colorScheme: "red",
+                textTransform: "capitalize",
+              }}
+              buttonText="Clear Data"
+              onClick={() => {
+                handleClear();
+              }}
```
[the remark this answers is not in the document]
Comment (jphechter) [reply services/ui-src/src/components/CompleteMeasureFooter/index.tsx:60]: Included.

File: services/ui-src/src/components/MeasureWrapper/index.tsx (hunk 3 above)
Comment (jphechter) [reply services/ui-src/src/components/MeasureWrapper/index.tsx:220]: Included.

**Review by benjaminpaige — DISMISSED** [review]
awesome

**Pull request merged**